### PR TITLE
Idex 2848

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java [hunk 1]
```diff
@@ -76,7 +77,8 @@
  */
 @Singleton
 public class MachineManager {
-    private static final Logger LOG = LoggerFactory.getLogger(MachineManager.class);
+    private static final Logger  LOG                          = LoggerFactory.getLogger(MachineManager.class);
+    private static final Pattern MACHINE_DISPLAY_NAME_PATTERN = Pattern.compile("^/?[a-zA-Z0-9_-]+$");
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: I think you should explain it [example](https://github.com/codenvy/che-core/blob/4.0/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java#L71)

File: platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        EnvironmentContext.setCurrent(envCont);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+        EnvironmentContext.reset();
+    }
+
+    @Test(expectedExceptions = MachineException.class)
+    public void shouldBeAnExceptionWhileValidatingDockerDisplayName() throws Exception {
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java]: i think that <code>shouldThrow</code> is better here

File: platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        final RecipeMachineCreationMetadata machineCreationMetadata = mock(RecipeMachineCreationMetadata.class);
+        final MachineRecipe machineRecipe = mock(MachineRecipe.class);
+        final InstanceProvider instanceProvider = mock(InstanceProvider.class);
+        final String type = "type";
+        final Set<String> recipeTypes = new HashSet<String>() {{
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java]: Hmm, actually here you create an instance of anonymous class which is not good, you can simplify it:
<code>new HashSet<>(Collections.singleton(type));

File: platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        envCont.setUser(new UserImpl("user", null, null, null, false));
+        EnvironmentContext.setCurrent(envCont);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+        EnvironmentContext.reset();
+    }
+
+    @Test(expectedExceptions = MachineException.class)
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java]: Consider adding expectedMessagePattern

File: platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java (hunk 3 above)
Comment (akorneta) [inline platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java]: :+1:

File: platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java (hunk 2 above)
Comment (akorneta) [inline platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/MachineManagerTest.java]: yeap

**Comment by garagatyi** [comment]
ok

**Comment by evoevodin** [comment]
:+1:

**Pull request merged**